.github/workflows/ruby_test.yml

```diff
@@ -0,0 +1,218 @@
+name: Ruby Test
```

Comment:
There's quite a lot of repetition in here isn't there? I wonder if we could refactor any of it out into a github actions repository or something

Reply:
We could possibly create our own actions to use. Haven't really looked into exactly what they look like, or indeed how-powerful they are.
Another possibility, which is a little simpler, would just be to use some of the inheritance mark-up built into yaml; although I did spot a suggestion online that the GH actions don't support some of the fancier yaml syntax.

Reply:
Yep, just quickly tested it, and yaml anchors aren't supported, so can't simplify that way.